qiskit/synthesis/clifford/clifford_decompose_layers.py

```diff
@@ -203,24 +204,25 @@ def _create_graph_state(cliff, validate=False):
     """
 
     num_qubits = cliff.num_qubits
-    rank = _compute_rank(cliff.stab_x)
+    rank = compute_rank((cliff.stab_x).astype(bool))
     H1_circ = QuantumCircuit(num_qubits, name="H1")
     cliffh = cliff.copy()
 
     if rank < num_qubits:
         stab = cliff.stab[:, :-1]
-        stab = _gauss_elimination(stab, num_qubits)
+        stab = stab.astype(bool)
```

Comment:
Same comment as above:

```suggestion
        stab = stab.astype(bool, copy=False)
```

Reply:
if I change this, it makes the tests fail

Reply:
Looking more closely at the code this is because `gauss_elimination()` is mutating the array in place and this will be a reference to the slice of the input clifford, so making this suggestion is causing the input clifford to be modified. The goal of this `astype()` call then is to make a copy. I think you should make that explicit since the intent isn't super clear with using `astype()` to do the copy. I'd suggest just use `stab.copy()` or `np.asarray(stab, copy=True, dtype=bool)`.